Sources/NIOSSL/NIOSSLHandler.swift

```diff
@@ -391,6 +503,8 @@ public class NIOSSLHandler : ChannelInboundHandler, ChannelOutboundHandler, Remo
             context.fireUserInboundEventTriggered(TLSUserEvent.shutdownCompleted)
 
             switch targetCompleteState {
+            case .outputClosed:
+                break
```

Comment:
I think this is a relevant edge: if it happens, we end up immediately in full closed.